lib/errors/base-error.ts

```diff
@@ -8,10 +21,10 @@
  * The Base Error all Sequelize Errors inherit from.
  */
 class BaseError extends Error {
```

Comment:
Can this be an [abstract](https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-classes-and-members) class?  (might not be available in our lowest supported TS version, if not can you add a todo?)